### ISSUE TITLE
Clarification on CV procedure

### ISSUE THREAD

**Issue by epeters3** [xliuhw/NLU-Evaluation-Data#4]
Thank you for sharing this great dataset!

In your paper, I saw that you evaluated each system using 10-fold cross validation. I see in this repo that each fold has its own train/test split. What were the specifics of how the systems were trained and evaluated? For example on each fold:
- Were they trained on the `trainset`s of k-1 folds, then evaluated just on the `testset` of the kth fold?
- Were they trained just on the `trainset` of the single fold, then evaluated on the `testset` of that single fold?
- Were they trained on a union of the `trainset` and `testset` of k-1 folds, then evaluated on the union of the `trainset` and `testset` of the kth fold?

**Comment by xliuhw** [comment]
Thanks for your question! 

Sorry, I don't remember the details for this on the top of my head :-( 

I might dig it out some time later if I could find some time. Currently I am very busy for other projects.

**Comment by epeters3** [comment]
That's ok @xliuhw don't worry. I was interested in reproducing the evaluation procedure you used exactly so we could compare our intent classification system to the ones in your paper. I appreciate your response. Good luck with your other projects! And thank you again for sharing this dataset.

Just so you know, we've recently included your great dataset in a large intent classification meta-dataset we've open sourced that's geared towards few-shot learning of intent classification tasks: https://github.com/bavard-ai/nlu-meta-dataset, just in case you were interested.

**Comment by kushalj001** [comment]
@epeters3 Did you figure out the training procedure? Would be great if you could share. I am looking for the same answers.

**Comment by epeters3** [comment]
Hi @kushalj001! Sorry I did not. I tried to deduce it from the code and the paper but wasn't quite able to.

**Comment by kushalj001** [comment]
Thank you @epeters3. Do you know how do we handle examples which do not have any annotated entities? I am training a DIET like multi-task model for intent classification and entity recognition. Should I label the tokens with 'O' for examples that have no entities? I am not sure how the DIET paper has handled this. Any insight would be helpful. Thank you.

**Comment by epeters3** [comment]
I think using 'O' for all tokens in that case sounds like a good idea. Looking at this paper, they used 3 annotators, deciding on entities with good annotator agreement. Since there were multiple annotators, I assume the risk is low of there being unlabeled entities in this dataset (label noise), so that should probably be a safe approach.

**Comment by epeters3** [comment]
And since you're using a dual Entity+Intent approach, I guess you'd still want to associate the intent label with the `__CLS__` token.

**Comment by kushalj001** [comment]
Yes, I am doing that. Thank you @epeters3 !

**Comment by xliuhw** [comment]
@epeters3 regarding your questions on CV training/evaluation procedure, sorry for unable to find time to dig it out earlier, now after in depth digging, I found what were done is the second one in your bullet point list:

For each K-Fold, they were trained just on the trainset of the single fold, then evaluated on the testset of that single fold.

Hope this helps!

Thanks for interested in our dataset and having included it into your meta-dataset!

**Comment by kushalj001** [comment]
I am running into a very weird issue. When I use the train and test set as given in each fold, my intent classifier overfits on the data. However, if I split the trainset internally into a new train and test set, the model works fine. The same thing happens with the test set. Moreover, if I combine the train and test set, shuffle the examples, and create a new train and test split, my model still works fine. 
This is making me think that the testing dataset for intent classification is out of distribution? Or there is a bug in my code. 
Let me know if you have any insights about this.
Thank you!

**Comment by epeters3** [comment]
Thanks @xliuhw for answering my question! That was kind of you to go digging for that answer. I hope the information will be useful for others in the future as well who are trying to reproduce your results or evaluate their system using the same setup as that used in your paper.
